shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1069,56 +1097,60 @@ - (void)keyboardWillBeHidden:(NSNotification*)notification {
   [self updateViewportMetrics];
 }
 
-- (void)dispatchPresses:(NSSet<UIPress*>*)presses API_AVAILABLE(ios(13.4)) {
+- (void)handlePressEvent:(NSSet<UIPress*>*)presses
+               withEvent:(UIPressesEvent*)event
+              nextAction:(void (^)())next API_AVAILABLE(ios(9.0)) {
   if (@available(iOS 13.4, *)) {
-    for (UIPress* press in presses) {
-      if (press.key == nil || press.phase == UIPressPhaseStationary ||
-          press.phase == UIPressPhaseChanged) {
-        continue;
-      }
-      NSMutableDictionary* keyMessage = [[@{
-        @"keymap" : @"ios",
-        @"type" : @"unknown",
-        @"keyCode" : @(press.key.keyCode),
-        @"modifiers" : @(press.key.modifierFlags),
-        @"characters" : press.key.characters,
-        @"charactersIgnoringModifiers" : press.key.charactersIgnoringModifiers
-      } mutableCopy] autorelease];
-
-      if (press.phase == UIPressPhaseBegan) {
-        keyMessage[@"type"] = @"keydown";
-      } else if (press.phase == UIPressPhaseEnded || press.phase == UIPressPhaseCancelled) {
-        keyMessage[@"type"] = @"keyup";
-      }
-
-      [[_engine.get() keyEventChannel] sendMessage:keyMessage];
+    bool __block wasHandled = false;
+    KeyEventCompleteCallback completeCallback =
+        ^void(bool handled, UIPress* press, UIPressesEvent* event) {
+          wasHandled = handled;
+          CFRunLoopStop(CFRunLoopGetCurrent());
```

Comment:
I wonder if it's possible to hide everything related to `CFRunLoopStop` into inside `FlutterKeyboardManager` because it's the kind of details I wish to split from the already complex `FlutterViewController`.

Reply:
Or can we expose `FlutterViewController::pressesBegan` and like as the public interface so that the callers don't have to repeat themselves as `nextAction`?

Reply:
Yes, I've moved it into `FlutterKeyboardManager`, but this API still exists in the `FlutterViewController`, since I wanted a single entrypoint that was accessible from the engine.